### PR TITLE
Azure: Fixes merging of requests with different Azure sources

### LINKED ISSUE #28114: Panel with queries from different sources not working
When a panel configured with query from different source in my case query A is application insights and query B is azure log anlytics. In query inspector no data is returned. It seems for different sources grafana is making two requests to 
/api/ds/query one for application insights and one for log anytics i can see in browser one query is getting blocked. Tried in both chrome and firefox same result chrome its showing cancelled. This was working fine before , i am not sure which version was running before as this was in kubernetes and when kubernetes cluster upgraded and restarted i think it downloaded a new container version. Attached the screenshot of network call from browser

Grafana version  v7.2.0

grafana

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
When I fixed #28114 with #28180 I didn't realize all the details of this data source. This PR tries to merge data from different Azure sources gracefully but as I don't know all the scenarios here I would appreciate help testing this. 

**Which issue(s) this PR fixes**:
Relates #28114

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by aknuds1** [comment]
@kylebrandt Are you the best person to help Hugo here?

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.ts [hunk 1]
```diff
@@ -98,16 +101,22 @@ export default class Datasource extends DataSourceApi<AzureMonitorQuery, AzureDa
...
+      return forkJoin(obs).pipe(
+        map((results: DataQueryResponse[]) => {
+          const data: DataFrame[] = [];
+          for (const result of results) {
+            for (const frame of result.data) {
+              data.push(frame);
+            }
+          }
+
+          return { state: LoadingState.Done, data };
```
Comment (hugohaggmark) [inline public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.ts:115]: The typings for this function `Observable<DataQueryResponseData>` seem incorrect when looking at what we are returning. But I just followed the return type that already existed in the code.

**Comment by hugohaggmark** [comment]
> @kylebrandt Are you the best person to help Hugo here?

Kyle found some issues with my previous fix so I add him too :)

**Review by kylebrandt — APPROVED** [review]
Works on my machine

image

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.ts (hunk 1 above)
Comment (ryantxu) [reply public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.ts:115]: weird -- it should be `Observable<DataDataQueryResponse>` I think, if that compiles and runs, we should change it

**Review by ryantxu — APPROVED** [review]
looks good, but I think we should fix the type if it does not break anythign

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.ts (hunk 1 above)
Comment (hugohaggmark) [reply public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.ts:114]: Yes, I fixed it

**Pull request merged**